### PR TITLE
class library: rename two PV ugens that are not

### LINKED ISSUE #2470: PV_JensenAndersen and PV_HainsworthFoote are misnomers
They are FFT analysis ugens, not PV ugens. They should be renamed, and the old names should be deprecated.

### PR DESCRIPTION
**Pull request by telephon**
The prefix PV_ is used for ugens that take an fft and return a new fft.
This fixes #2470.

### CODE REVIEW

File: SCClassLibrary/Common/Audio/FFT2.sc [hunk 1]
```diff
@@ -56,19 +56,23 @@ Convolution3 : UGen {
 
 
 //jensen andersen inspired FFT feature detector
-PV_JensenAndersen : PV_ChainUGen {
+JensenAndersen : PV_ChainUGen {
```
Comment (Sciss) [inline SCClassLibrary/Common/Audio/FFT2.sc]: Doesn't this end up with the wrong UGen name for the server? The UGen is still called `PV_JensenAndersen`, or not? Doesn't the class file then have to provide an explicit name somewhere?

**Comment by snappizz** [comment]
I don't think this will work. If you use JensenAndersen and HainsworthFoote, the server will try to create ugens of those names, which don't exist.

The change can't just be done in sclang, because we have to consider alternate clients.

**Comment by telephon** [comment]
Ah yes, my omission. This one will work, it just changes the interface for now.

**Comment by jamshark70** [comment]
> This one will work

No, it doesn't. After adding, by hand, `HainsworthFoote : PV_HainsworthFoote {}` :

```
(
SynthDef(\fftod, {
    var source1, detect;
    source1= LFSaw.ar(LFNoise0.kr(1,90,400),0,0.5);
    detect= HainsworthFoote.ar(FFT(LocalBuf(2048),source1), 1.0, 0.0, 0.9, 0.5);
    Out.ar(0,Pan2.ar(source1,-1.0)+ Pan2.ar(SinOsc.ar(440,0,Decay.ar(0.1*detect,0.1)),1.0));
}).play(s);
)

->

exception in GraphDef_Recv: UGen 'HainsworthFoote' not installed.
```

... which is exactly what both Nathan and sciss said would happen, but which was subsequently ignored.

The reason for the failure is that the currently-committed approach is not a true redirect. If `HainsworthFoote` is a subclass of `PV_HainsworthFoote` that adds no other functionality, then `HainsworthFoote.ar(...)` will give you an instance of `HainsworthFoote` and **not** of `PV_HainsworthFoote` -- and then, the `UGen:name` written into the SynthDef will be `HainsworthFoote`, and this does not exist in the server.

You could make it a true redirect:

```
HainsworthFoote : PV_HainsworthFoote {
	*ar { arg buffer, proph=0.0, propf=0.0, threshold=1.0, waittime=0.04;
		^PV_HainsworthFoote.ar(buffer, proph, propf, threshold, waittime)
	}
}
```

(note that `^super.ar` would not be OK here)...

Or, you could override `name`:

```
HainsworthFoote : PV_HainsworthFoote {
	name { ^"PV_HainsworthFoote" }
}
```

Or, remove `PV_HainsworthFoote` altogether (note also the correction to the superclass):

```
HainsworthFoote : UGen {
	*ar { arg buffer, proph=0.0, propf=0.0, threshold=1.0, waittime=0.04;
		^PV_HainsworthFoote.ar(buffer, proph, propf, threshold, waittime)
	}
	name { ^"PV_HainsworthFoote" }
}
```

I would suggest closing this PR and submitting a new one with a clean revision history.

**Comment by telephon** [comment]
Good to know, thank you for the careful explanation. If you all agree that this is the way we should do it, I can do that.

**Pull request closed without merging**